subcommands/version/update_self_unix.go

```diff
@@ -0,0 +1,17 @@
+//go:build !windows
+// +build !windows
+
+package version
+
+import (
+	"io/fs"
+	"os"
+)
+
+func updateSelf(exe string, buff []byte, mode fs.FileMode) error {
+	tmp := exe + ".tmp"
+	if err := os.WriteFile(tmp, buff, mode); err != nil {
+		return err
+	}
+	return os.Rename(tmp, exe)
```

Comment:
I am still surprised that it works like that in linux. Maybe it works in 99% and some times may lead to crashes.

Reply:
afaik, not whole executable content is loaded into memory during process startup, it's memory mapped, and data are loaded and written on demand.
At least that is what I remember/learned around 15 years ago. Maybe it's only related only to shared libraries...

Reply:
I've forgotten about inode and renaming stuff, hence it works on linux systems.

Reply:
Yep, on linux you can run `rm -rf /` and the system will continue to work for quite some time.

Reply:
OK. Let me test it on my dev laptop :)